source/slang/check.cpp

```diff
@@ -6073,9 +6118,9 @@ namespace Slang
                 case 'w': case 'a': elementIndex = 3; break;
                 default:
                     // An invalid character in the swizzle is an error
-                    if (!isRewriteMode())
+                    if (!isRewriteMode() && !anyError)
                     {
-                        getSink()->diagnose(swizExpr, Diagnostics::unimplemented, "invalid component name for swizzle");
```

Comment:
While it is good to turn these into real errors rather than `unimplemented` errors, it would be good to retain the more detailed messages here. If the user writes `foo.xyd` it is better to tell them `invalid component name 'd' for swizzle` and not just `invalid swizzle string 'xyd' on type 'float4'`. Similar for `swizzle component 'z' out of range for type 'float2'`.

These are minor points in a larger change, but we also might want to get into the habit of pulling pure cleanup work into their own PRs so that the inevitable bike-shedding doesn't happen on major feature PRs. :)

Reply:
Actually I don't think that is a better error message.
Usually when you are in this situation, you are treating the memberbase as a wrong type. The issue I ran into is that I have a `float rs` and uses `rs.baseColor` treating `rs` as a struct instead of a float. Had I know `rs` is a `float` I wouldn't be writing `baseColor` at all. In this case, an error message telling me `s` is not a valid component is not as informative as `baseColor` is an invalid swizzle string.

Reply:
Good point. Then I agree we should leave it as-is then, except maybe don't call it a "swizzle string" - I don't know what a better word is but "string" doesn't feel right given that we don't write `foo."xyz"`.

Reply:
changed it to 'swizzle pattern'